### DISCUSSION TITLE
Is there any way to translate the Admin UI?

### DISCUSSION THREAD

**Discussion by DenMarty** [medusajs/medusa#9831] [Q&A] [answered]
Based on this [#2003](https://github.com/medusajs/medusa/discussions/2003) discussion, it's clear that things are very difficult with Admin UI translations at the moment. Looking through the documentation, I found a lot of useful information, but not a word about translating the UI (although the UI settings themselves seem to provide for such a feature).

Finally, a question has matured: 

- Is there a third-party way to implement translating the Admin Dashboard into another language? 

- Maybe there is an example of integrating i18 or similar tool?

**Comment by kasterby** [comment, marked as answer]
@DenMarty It looks like the discussion you linked is about store-wide translations, not only the admin dashboard.
Translating the admin dashboard seems quite straightforward: https://github.com/medusajs/medusa/tree/develop/packages/admin/dashboard/src/i18n
Adding a new lang in there should make it available under the language configuration dropdown in the admin UI account settings
Hope that helps!

**Reply by DenMarty** [reply]
Thanks, I really hadn't paid attention to that. I think it would be useful to add information about it to the documentation.

**Reply by Ksztof** [reply]
I don’t know why it took me so long to find a way to translate the admin panel, but it actually works with version 2.8.3. There really should just be a link to this repo in the Medusa documentation, but instead it’s hidden away—like most things in Medusa, to be honest :( Anyway, you saved me, thank you!

**Comment by jabarientergalactic** [comment]
[ru.json](https://github.com/user-attachments/files/32567556/ru.json)
We noticed 2.21.1 added Finnish and Swedish translations to the admin
dashboard — great to see the language list growing.

We've been running a full Russian translation of the admin dashboard in
production for a while (as a client-side overlay, since there wasn't an
official ru.json yet) and would like to contribute it upstream following
the process described in the admin translation contribution guide (copying
en.json to ru.json, translating all keys, registering the language in
translations/index.ts and languages.ts).

Attaching our current ru.json as a starting point — is there anything
special we should know before opening a PR (e.g. pluralization rules for
Slavic languages, review timeline, anyone already working on a Russian
translation we should coordinate with)?
